src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -830,6 +832,8 @@ private bool TryComputeWeightedDistance(BlockSyntax leftBlock, BlockSyntax right
                     if (leftCatch.Declaration == null && leftCatch.Filter == null &&
                         rightCatch.Declaration == null && rightCatch.Filter == null)
                     {
+                        RoslynDebug.Assert(leftCatch.Parent is object);
+                        RoslynDebug.Assert(rightCatch.Parent is object);
```

Comment:
I think it'd be better to consistently use Parent! rather than adding asserts in some places and ! in others.

Reply:
Going to take this suggestion as this is primarily your area. But I think the IDE team should come to an agreement on what they believe the preferred approach should be. I've gotten the opposite feedback from others. 

I'm mostly ambivilent

---
In reply to: [362902287](https://github.com/dotnet/roslyn/pull/40699#discussion_r362902287) [](ancestors = 362902287)

Reply:
We can discuss it at the IDE design meeting. I wonder if we need some broader discussion on what we generally recommend to our customers though. 

---
In reply to: [362904328](https://github.com/dotnet/roslyn/pull/40699#discussion_r362904328) [](ancestors = 362904328,362902287)

Reply:
I also think this is area-dependent. Code that validates that the syntax tree is well-formed and then does analysis on it is different from code that did not make the up-front validation.

---
In reply to: [362905900](https://github.com/dotnet/roslyn/pull/40699#discussion_r362905900) [](ancestors = 362905900,362904328,362902287)